sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala

```diff
@@ -433,13 +493,38 @@ object ScalaJSPlugin extends AutoPlugin {
         "List of arguments to pass to the Scala.js Java System.properties.",
         CTask)
 
+    val scalaJSSourceFiles = AttributeKey[Seq[File]]("scalaJSSourceFiles",
+        "Files used to compute this value (can be used in FileFunctions later).",
+        KeyRanks.Invisible)
+
     val scalaJSSourceMap = AttributeKey[File]("scalaJSSourceMap",
         "Source map file attached to an Attributed .js file.",
         BSetting)
   }
 
   import autoImport._
-  import ScalaJSPluginInternal._
+
+  /** Maps a [[Stage]] to the corresponding [[sbt.Def.TaskKey TaskKey]].
+   *
+   *  For example, [[Stage.FastOpt]] (aka `FastOptStage`) is mapped to
+   *  [[fastOptJS]].
+   */
+  val stageKeys: Map[Stage, TaskKey[Attributed[File]]] = Map(
+      Stage.FastOpt -> fastOptJS,
+      Stage.FullOpt -> fullOptJS
+  )
+
+  /** Logs the current statistics about the global IR cache. */
+  def logIRCacheStats(logger: Logger): Unit =
+    logger.debug("Global IR cache stats: " + globalIRCache.stats.logLine)
+
+  /** Configures the given `IncOptions` for use with Scala.js.
+   *
+   *  This ensures that .sjsir files are pruned as needed in incremental
+   *  compiler runs.
+   */
+  def configureIncOptionsForScalaJS(incOptions: IncOptions): IncOptions =
+    SBTCompat.scalaJSPatchIncOptions(incOptions)
```

Comment:
This is in case someone completely overrides their `incOptions` with `:=` instead of transforming it and preserving our external hooks. In that case they'll want to install our external hook on their brand new `IncOptions`.

That said, now that I think about it more, it's probably extremely niche. It's probably better to just not expose it now, and do so if someone ever asks for it.